pkg/kubectl/cmd/context.go

```diff
@@ -0,0 +1,31 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+  "io"
+)
+
+type CmdCustomizations struct {
+  baseCmd       string
```

Comment:
Other than `baseCmd` and `cmdName` you may also want to have `fullName`. Sometimes you have commands that live in the third or below level of the tree, for example `kubectl create quota`, in which case just the base and cmd name are not enough.